### PR TITLE
feat: add s3 module to owasp-zap IaC and fix CI

### PR DESCRIPTION
**Pull request by CalvinRodo**
# Summary | Résumé

- Update makefile so that it includes scanners/owasp-zap and terragrunt
- add ci linting and fmt steps
- fix issues found by linter and fmt that weren't surfacing because CI wasn't running properly.

### CODE REVIEW

File: terragrunt/aws/scanners/owasp-zap/s3.tf [hunk 1]
```diff
@@ -1,28 +1,17 @@
...
-  acl    = "private"
-  server_side_encryption_configuration {
-    rule {
-      apply_server_side_encryption_by_default {
-        sse_algorithm = "AES256"
-      }
-    }
-  }
+locals {
+  owasp_zap_report_data_name = "${var.product_name}-${var.env}owasp-zap-report-data"
```
Comment (patheard) [inline terragrunt/aws/scanners/owasp-zap/s3.tf:2]: Minor nit - hyphen between `var.env` and bucket name:
```hcl
"${var.product_name}-${var.env}-owasp-zap-report-data"
```

**Review by patheard — COMMENTED** [review]
Minor s3 bucket name comment.

Apart from the lint failure, changes look good!

File: Makefile [hunk 2]
```diff
@@ -14,6 +16,7 @@ help:
 	@echo "  dev         -- Run API dev server"
 	@echo "  format      -- Alias for fmt"
 	@echo "  fmt         -- Run formatters"
+	@echo "  fmt-ci      -- Run formatters in check mode for CI"
 	@echo "  install     -- Install needed Python and NPM libraries"
 	@echo "  install-dev -- Installs needed development libraries"
 	@echo "  lint        -- Run linters"
```
Comment (dj2) [inline Makefile:22]: nit: add lint-ci

**Review by mohamed-cds — APPROVED** [review]
Nice! 🚀 ✅

File: api/Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-.PHONY: dev fmt install lint migrations test
+.PHONY: dev fmt install lint migrations test fmt-ci
```
Comment (dj2) [inline api/Makefile:1]: nit: lint-ci

File: Makefile (hunk 2 above)
Comment (patheard) [inline Makefile:22]: Super minor - `lint-ci` help blurb.

File: terragrunt/Makefile [hunk 4]
```diff
@@ -1,7 +1,21 @@
-.PHONY: fmt
+.PHONY: fmt checkov install lint test fmt-ci lint-ci
```
Comment (dj2) [inline terragrunt/Makefile:1]: nit: build install-dev

**Review by dj2 — APPROVED** [review]
LGTM, just a few nits.

File: api/Makefile (hunk 3 above)
Comment (patheard) [inline api/Makefile:1]: `lint-ci` rule for PHONY.

**Review by patheard — COMMENTED** [review]
Just a few minor comments about Makefiles targets.

**Pull request merged**